Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -17,6 +17,100 @@ import NIO
 let crlf: StaticString = "\r\n"
 let headerSeparator: StaticString = ": "
 
+/* private but tests */ internal struct HTTPNonContiguousHeaderCommaSeparatedValuedFieldHandler
```

Comment:
I think this can be more general and made public: we can call this a `HTTPListHeaderIterator`.